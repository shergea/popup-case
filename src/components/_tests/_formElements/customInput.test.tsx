import { Formik } from "formik";
import renderer from "react-test-renderer";
import CustomInput from "../../formElements/customInput";

it("renders correctly", () => {
  const fieldName = "firstName";
  const labelName = "First Name";
  const tree = renderer
    .create(
      <Formik
        validate={(values: any) => {
          const errors = {} as any;

          if (!values?.firstName) {
            errors.firstName = "Required.";
          }

          return errors;
        }}
        initialValues={{}}
        onSubmit={() => {}}
      >
        <CustomInput
          id={fieldName}
          name={fieldName}
          label={labelName}
          placeholder={labelName}
        />
      </Formik>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
