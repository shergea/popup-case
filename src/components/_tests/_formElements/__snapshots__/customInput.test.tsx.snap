// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mt-5"
>
  <label
    className="block mb-2 text-m font-medium text-gray-900 dark:text-gray-300"
    htmlFor="firstName"
  >
    First Name
  </label>
  <input
    className="border border-gray-300 text-gray-900 text-m rounded-lg block w-full p-2.5 outline-0"
    id="firstName"
    name="firstName"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="First Name"
    type="text"
  />
</div>
`;
